contracts/crosschain/wormhole/WormholeGatewayAdapter.sol

```diff
@@ -0,0 +1,296 @@
+        require(_remoteGateways[chainId] == address(0), RemoteGatewayAlreadyRegistered(chainId));
+        _remoteGateways[chainId] = addr;
+        emit RegisteredRemoteGateway(chainId, addr);
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function supportsAttribute(bytes4 selector) public pure returns (bool) {
+        return selector == IERC7786Attributes.requestRelay.selector;
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function sendMessage(
+        bytes calldata recipient, // Binary Interoperable Address
+        bytes calldata payload,
+        bytes[] calldata attributes
+    ) external payable returns (bytes32 sendId) {
+        bool relay = false;
+        uint256 value = msg.value;
+        uint256 gasLimit = 0;
+
+        for (uint256 i = 0; i < attributes.length; ++i) {
+            (relay, value, gasLimit, ) = ERC7786Attributes.tryDecodeRequestRelayCalldata(attributes[i]);
+            require(relay, UnsupportedAttribute(attributes[i].length < 0x04 ? bytes4(0) : bytes4(attributes[i])));
+        }
+
+        if (relay) {
+            // TODO: Do we care about the returned "sequence"?
+            bytes memory encoded = abi.encode(
+                sendId,
+                InteroperableAddress.formatEvmV1(block.chainid, msg.sender),
+                recipient,
+                payload
+            );
+            _wormholeRelayer.sendPayloadToEvm{value: msg.value}(
+                getWormholeChain(recipient),
+                getRemoteGateway(recipient),
+                encoded,
+                value,
+                gasLimit
+            );
+        } else {
+            sendId = bytes32(++_lastSendId);
+
+            // Note: this reverts with UnsupportedChainId if the recipient is not on a supported chain.
+            // No real need to check the return value.
+            getRemoteGateway(recipient);
+
+            _pending[sendId] = PendingMessage(true, msg.sender, value, recipient, payload);
+        }
+
+        emit MessageSent(
+            sendId,
+            InteroperableAddress.formatEvmV1(block.chainid, msg.sender),
+            recipient,
+            payload,
+            value,
+            attributes
+        );
+    }
+
+    /// @dev Returns a quote for the value that must be passed to {requestRelay}
+    function quoteRelay(
+        bytes calldata recipient, // Binary Interoperable Address
+        bytes calldata /*payload*/,
+        bytes[] calldata /*attributes*/,
+        uint256 value,
+        uint256 gasLimit,
+        address /*refundRecipient*/
+    ) external view returns (uint256) {
+        (uint256 cost, ) = _wormholeRelayer.quoteEVMDeliveryPrice(getWormholeChain(recipient), value, gasLimit);
+        return cost - value;
+    }
+
+    /// @dev Relay a message that was initiated by {sendMessage}.
+    function requestRelay(bytes32 sendId, uint256 gasLimit, address /*refundRecipient*/) external payable {
```

Comment:
Why is `refundRecipient` ignored? There's a version of `sendPayloadToEvm` that takes a refund address.

https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/575181b586a315d8f9813eab82e4cb98b45bc381/src/interfaces/IWormholeRelayer.sol#L123-L131

Reply:
changed